pkg/migrator/migrator.go

```diff
@@ -39,115 +39,141 @@ func NewMigrator(client client.Client) migrator {
 	return migrator{
 		logger: log.NewEntry(log.New()),
 		client: client,
+		retry:  3,
 	}
 }
 
 type migrator struct {
 	logger *logrus.Entry
 	client client.Client
+	retry  int
 }
 
-// Migrate updates existing Subscriptions, deletes the CSCs installed during
+// doMigrate updates existing Subscriptions, deletes the CSCs installed during
 // operator installation as well the stale datastore CSCs when the cluster is
-// migrating from Openshift 4.1.x to Openshift 4.2.x
+// migrating from Openshift 4.1.z to Openshift 4.2.z
+func (m *migrator) doMigrate(operatorNamespace string) []string {
+	installedCscs, errorList := m.updateSubscriptions()
+	if errorList != nil {
+		return errorList
```

Comment:
I am wondering if we should collect the list of errors here too and return it at the end. For that we would need to ensure that the errors in one step will not affect the next.